src/util/misc.js

```diff
@@ -221,50 +237,185 @@
     projectStrokeOnPoints: function (points, options, openPath) {
       var coords = [], s = options.strokeWidth / 2,
           strokeUniformScalar = options.strokeUniform ?
-            new fabric.Point(1 / options.scaleX, 1 / options.scaleY) : new fabric.Point(1, 1),
-          getStrokeHatVector = function (v) {
-            var scalar = s / (Math.hypot(v.x, v.y));
-            return new fabric.Point(v.x * scalar * strokeUniformScalar.x, v.y * scalar * strokeUniformScalar.y);
-          };
+            new fabric.Point(1 / options.scaleX, 1 / options.scaleY) : new fabric.Point(1, 1);
+
       if (points.length <= 1) {return coords;}
+
       points.forEach(function (p, index) {
         var A = new fabric.Point(p.x, p.y), B, C;
         if (index === 0) {
           C = points[index + 1];
-          B = openPath ? getStrokeHatVector(fabric.util.createVector(C, A)).addEquals(A) : points[points.length - 1];
+          B = openPath ? A : points[points.length - 1];
         }
         else if (index === points.length - 1) {
           B = points[index - 1];
-          C = openPath ? getStrokeHatVector(fabric.util.createVector(B, A)).addEquals(A) : points[0];
+          C = openPath ? A : points[0];
         }
         else {
           B = points[index - 1];
           C = points[index + 1];
         }
-        var bisector = fabric.util.getBisector(A, B, C),
-            bisectorVector = bisector.vector,
+
+        if (openPath && index === 0) {
+          var scaledA = new fabric.Point(A.x * options.scaleX, A.y * options.scaleY),
+              scaledC = new fabric.Point(C.x * options.scaleX, C.y * options.scaleY);
+
+          var vector = fabric.util.createVector(
+                        options.strokeUniform ? scaledA : A,
+                        options.strokeUniform ? scaledC : C
+                      ),
+              hatOrthogonalVector = fabric.util.getOrthogonalUnitVector(vector),
+              orthogonalVector = new fabric.Point(
+                hatOrthogonalVector.x * s * strokeUniformScalar.x, 
+                hatOrthogonalVector.y * s * strokeUniformScalar.y 
+              );
+
+          coords.push(A.add(orthogonalVector));
+          coords.push(A.subtract(orthogonalVector));
+          return;
+        }
+
+        if (openPath && index === points.length - 1) {
```

Comment:
This is redundant
Tet's merge the 2 cases `openPath && (index === 0 || index === points.length - 1)`
We can do something like
```js
var D = index === 0 ? C : B

```